test/000_fullchain.test.ts

```diff
@@ -126,36 +127,34 @@ describe('Integration tests', function () {
             PocoMode.CLASSIC,
         );
         const schedulerRewardPerTask = workerpoolPrice - workerRewardPerTask;
-        // Check initial balances.
-        // TODO save initial balances and use them in for loop for comparison.
-        await checkBalancesAndFrozens({
-            proxyBalance: dealPrice + schedulerStakePerDeal,
-            accounts: [
-                { signer: sponsor, balance: 0, frozen: dealPrice },
-                { signer: requester, balance: 0, frozen: 0 },
-                { signer: scheduler, balance: 0, frozen: schedulerStakePerDeal },
-                { signer: appProvider, balance: 0, frozen: 0 },
-                { signer: datasetProvider, balance: 0, frozen: 0 },
-                { signer: worker1, balance: 0, frozen: 0 },
-            ],
-        });
+        // Save frozens
+
+        const accounts = [sponsor, requester, scheduler, appProvider, datasetProvider, ...workers];
+        const accountsInitialFrozens = await getInitialFrozens(accounts);
         // Finalize each task and check balance changes.
         for (let taskIndex = 0; taskIndex < volume; taskIndex++) {
             const taskId = await iexecWrapper.initializeTask(dealId, taskIndex);
-            const { workerStakePerTask } = await iexecWrapper.contributeToTask(
-                dealId,
-                taskIndex,
-                callbackResultDigest,
-                worker1,
-            );
-            await iexecPoco
-                .connect(worker1)
-                .reveal(taskId, callbackResultDigest)
-                .then((tx) => tx.wait());
-            await iexecPoco
+            const workerStakePerTask = await iexecPoco
+                .viewDeal(dealId)
+                .then((deal) => deal.workerStake.toNumber());
```

Comment:
This can go outside of the `for` loop.

Reply:
[set workerStakePerTask outside of the loop](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/aeeba74c359210e08fca73f8efe80cdf2d148475)